tests/rados/test_osd_async_recovery.py

```diff
@@ -0,0 +1,143 @@
+        for entry in target_configs.values():
+            log.debug(
+                f"Creating {entry['pool_type']} pool on the cluster with name {entry['pool_name']}"
+            )
+            if entry.get("pool_type", "replicated") == "erasure":
+                method_should_succeed(
+                    rados_object.create_erasure_pool, name=entry["pool_name"], **entry
+                )
+            else:
+                method_should_succeed(
+                    rados_object.create_pool,
+                    **entry,
+                )
+            rados_object.bench_write(
+                pool_name=entry["pool_name"], rados_write_duration=600, background=True
+            )
+
+            log_lines = [
+                "choose_async_recovery_replicated candidates by cost are",
+                "choose_async_recovery_replicated",
+            ]
+            osd_list = []
+            for node in ceph_nodes:
+                if node.role == "osd":
+                    node_osds = rados_object.collect_osd_daemon_ids(node)
+                    osd_list = osd_list + node_osds
+            log.info(f"The number of OSDs in the cluster are-{len(osd_list)}")
+
+            log_osd_count = 0
+            for osd_id in osd_list:
+                init_time, _ = installer.exec_command(
+                    cmd="sudo date '+%Y-%m-%d %H:%M:%S'"
+                )
+                if not rados_object.change_osd_state(action="restart", target=osd_id):
+                    log.error(f" Failed to restart OSD : {osd_id}")
+                    return 1
+                method_should_succeed(wait_for_clean_pg_sets, rados_object)
+                end_time, _ = installer.exec_command(
+                    cmd="sudo date '+%Y-%m-%d %H:%M:%S'"
+                )
+                if verify_async_recovery_log(
+                    rados_obj=rados_object,
+                    osd=osd_id,
+                    start_time=init_time,
+                    end_time=end_time,
+                    lines=log_lines,
+                ):
+                    log.info(f"found the log lines in the osd id -{osd_id}")
+                    log_osd_count = log_osd_count + 1
+                    if log_osd_count == 2:
+                        break
+                log.info("Found the log lines")
+
+            if log_osd_count == 0:
+                log.error("Log lines not found in any of the OSDs")
+                return 1
+    except Exception as e:
+        log.info(e)
+        log.info(traceback.format_exc())
+        return 1
+    finally:
+        log.info("Execution of finally block")
+        if config.get("delete_pool"):
+            method_should_succeed(rados_object.detete_pool, entry["pool_name"])
+            log.info("deleted the pool successfully")
+        mon_obj.remove_config(section="osd", name="osd_async_recovery_min_cost")
+        mon_obj.remove_config(section="osd", name="debug_osd")
+        # intentional wait for 5 seconds
+        time.sleep(5)
+
```

Comment:
There is no pass condition in the test. please add return 0 at the end after every check is passed.

Reply:
Pass condition included.